### PR TITLE
Copy ome.api.* documentation to omero.api.* (1)

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/KLCaZX4X/19-copy-ome-api-documentation-to-omero-api

This PR initializes the copy of the Javadoc documentation from the first set of classes of `ome.api.*` classes into the slice classes.

To review this PR, either build the slice documentation via `slice2html` or use the published CI documentation. Check the links are all working and there is no obvious error.

### CODE REVIEW

**Comment by ximenesuk** [comment]
@sbesson sorry, I meant to look at this last week but failed to find the published CI docs and then forgot about it. Where are the published docs for the latest merge build?

**Comment by sbesson** [comment]
@ximenesuk: see https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/slice2html/  or alternatively download the OMERO.docs artifacts which contains the API documentation.

**Comment by ximenesuk** [comment]
@sbesson thanks, I was looking in all the wrong places - so long since I have had to look on Jenkins. I'll review tomorrow.

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 1]
```diff
@@ -18,52 +18,427 @@ module omero {
...
+                 * modified by the user's umask settings.
+                 * If no permissions is set, the default will be
+                 * {@link omero.model.Permissions#USER_PRIVATE},
+                 * i.e. a group in which no user can see the other group
+                 * member's data.
+                 *
+                 * @param group  a new
+                 * {@link omero.model.ExperimenterGroup} instance. Not null.
+                 * @return id of the newly created {@link ExperimenterGroup}
+                 * @see <a href="https://trac.openmicroscopy.org/ome/ticket/1434">ticket:1434"</a>
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: I know this is a copy across from ome but do you want to pick up minor fixes here? If so then there is a stray double quote:

ticket:1434" -> ticket:1434

Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: More generally, none of the trac links resolve to anything. Is that expected in this build? The source looks like the periods in the url are represented as double colons. Is that some intermediate representation or is it broken?

File: components/blitz/resources/omero/api/IContainer.ice [hunk 2]
```diff
@@ -19,25 +19,420 @@ module omero {
...
+                 * rootNodeType, any of its containees or an
+                 * {@link omero.model.Image}.
+                 * </p>
+                 * <p>
+                 * For example, say that you pass in the ids of six Images:
+                 * <code>i1, i2, i3, i4, i5, i6</code>.
+                 * If the P/D/I hierarchy in the DB looks like this:
+                 * </p>
+                 *
+                 * <pre>
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IContainer.ice]: The <pre> block in the rendered doc is not well formatted, see: https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/slice2html/omero/api/IContainer.html#findContainerHierarchies

File: components/blitz/resources/omero/api/IContainer.ice [hunk 3]
```diff
@@ -19,25 +19,420 @@ module omero {
...
+                 *            Timestamp.valueOf("YYYY-MM-DD hh:mm:ss.ms");
+                 *            <p>
+                 *            <code>limit</code> and <code>offset</code> are
+                 *            applied at the Image-level. That is, calling
+                 *            with Dataset.class, limit == 10 and offset == 0
+                 *            will first perform one query to get an effective
+                 *            set of rootNodeIds, then getImages will be
+                 *            called with an effective rootNodeType of
+                 *            Image.class and the new ids.
+                 *            </p>.
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IContainer.ice]: A stray period which renders onto the next lin:

<p>. -> <p>

File: components/blitz/resources/omero/api/ILdap.ice [hunk 4]
```diff
@@ -16,26 +16,163 @@ module omero {
...
+                 * for attribute objectClass = person.
+                 *
+                 * @return all Experimenter list.
+                 */
                 idempotent ExperimenterList searchAll() throws ServerError;
+
+                /**
+                 * Searches Distinguished Name in groups
+                 *
+                 * @param attr   Name of member attribute. Never null or empty.
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/ILdap.ice]: This param marker is spilling into the short comment for this method:

https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/slice2html/omero/api/ILdap.html#ILdap

this is happening for most of the methods here, part of the first param is appearing with the @param being explicitly quoted.

File: components/blitz/resources/omero/api/ILdap.ice [hunk 5]
```diff
@@ -16,26 +16,163 @@ module omero {
...
+                 * @return       List of groups which contains DN.
+                 */
                 idempotent StringSet searchDnInGroups(string attr, string value) throws ServerError;
+
+                /**
+                 * Searches all {@link omero.model.Experimenter} in LDAP for
+                 * specified attribute
+                 *
+                 * @param dn        Distinguished Name base for search. Never
+                 *                  null.
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/ILdap.ice]: ditto

File: components/blitz/resources/omero/api/ILdap.ice [hunk 6]
```diff
@@ -16,26 +16,163 @@ module omero {
...
+                 */
                 idempotent ExperimenterList searchByAttribute(string dn, string attribute, string value) throws ServerError;
+
+                /**
+                 * Searches all {@link omero.model.Experimenter} in LDAP for
+                 * specified attributes. Attributes should be specified in
+                 * StringSet and their values should be set in equivalent
+                 * StringSet.
+                 *
+                 * @param dn         Distinguished Name base for search. Never
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/ILdap.ice]: ditto

File: components/blitz/resources/omero/api/ILdap.ice [hunk 7]
```diff
@@ -16,26 +16,163 @@ module omero {
...
+                 *                   or empty.
+                 * @return           List of Experimenters.
+                 */
                 idempotent ExperimenterList searchByAttributes(string dn, StringSet attributes, StringSet values) throws ServerError;
+
+                /**
+                 * Searches one {@link omero.model.Experimenter} in LDAP for
+                 * specified Distinguished Name
+                 *
+                 * @param userdn
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/ILdap.ice]: ditto

**Comment by ximenesuk** [comment]
The copying on the whole looks fine. Some of my comments relate to minor errors that were in the original `ome` source. Other comments may well be because this isn't a fully rendered doc with css etc?

**Comment by sbesson** [comment]
@ximenesuk: commits pushed to address your comments. A couple of comments:
- it looks like `@see` does not cope well with external hyperlinks so I moved the Trac ticket references to the method descriptions instead
- the `@param` leakage into the short description seemed to be fixed by adding final dots to the method description consistently with your observation in https://github.com/openmicroscopy/openmicroscopy/pull/4547#discussion_r58495893.

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 8]
```diff
@@ -18,52 +18,428 @@ module omero {
...
+                 * {@link omero.model.FileAnnotation} with
+                 * the namespace:
+                 *  "openmicroscopy.org/omero/experimenter/photo"
+                 * (NSEXPERIMENTERPHOTO).
+                 * If such an {@link omero.model.OriginalFile} instance
+                 * already exists, it will be overwritten. If more than one
+                 * photo is present, the oldest version will be modified (i.e.
+                 * the highest updateEvent id).
+                 *
+                 * Note: as outlined in ticket:1794, this photo will be placed
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: This ticket reference needs to be a trac url

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 9]
```diff
@@ -18,52 +18,428 @@ module omero {
...
+                 *         no default group is found, an exception will be
+                 *         thrown.
+                 */
                 idempotent omero::model::ExperimenterGroup getDefaultGroup(long experimenterId) throws ServerError;
+
+                /**
+                 * Looks up {@link omero.model.Experimenter} experimenters who
+                 * use LDAP authentication  (has set dn on password table).
+                 *
+                 * @return Experimenter. Never null.
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: There are quite a few cases where the documented param/return does not match the method param/return in various ways. Here the documented return type is an Experimenter but a string is returned - experimenter name? Should those be fixed here in this PR?

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 10]
```diff
@@ -18,52 +18,428 @@ module omero {
...
                 idempotent RList lookupLdapAuthExperimenters() throws ServerError;
+
+                /**
+                 * Finds the ids for all groups for which the given
+                 * {@link omero.model.Experimenter} is a member.
+                 *
+                 * @param e Non-null, managed (i.e. with id)
+                 * @see omero.model.Details#getOwner
+                 */
                 idempotent LongList getMemberOfGroupIds(omero::model::Experimenter exp) throws ServerError;
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: There are also quite a few places where no return is documented. Again, is this something to be fixed now?

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 11]
```diff
@@ -18,52 +18,428 @@ module omero {
...
+
+                /**
+                 * Finds the ids for all groups for which the given
+                 * {@link omero.model.Experimenter} is owner/leader.
+                 *
+                 * @param e Non-null, managed (i.e. with id)
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: Here the documented parameter is `e`, the actual parameter is `exp`.

File: components/blitz/resources/omero/api/IAdmin.ice [hunk 12]
```diff
@@ -18,52 +18,428 @@ module omero {
...
+                 *
+                 * @param group the ExperimenterGroup to update.
+                 */
                 void updateGroup(omero::model::ExperimenterGroup group) throws ServerError;
+
+                /**
+                 * Creates and returns a new user. This user will be created
+                 * with the default group specified.
+                 *
+                 * @param newUser a new {@link omero.model.Experimenter}
```
Comment (ximenesuk) [inline components/blitz/resources/omero/api/IAdmin.ice]: Final example, @param is `newUser` actual parameter is `experimenter`

**Comment by ximenesuk** [comment]
The four examples above are more of a question. The documentation doesn't match the code, should this be fixed  as the move is made from ome -> omero or left for a later sweep?

**Comment by sbesson** [comment]
@ximenesuk: I would be in favor or addressing only the mismatching `@param PARAMETER` here and leave the content/description in sync as we should review it together with `ome.api.*`. I can push a commit to fix the mismatching property names and the Trac ticket in this PR.

**Comment by ximenesuk** [comment]
Looks good to merge.

**Comment by joshmoore** [comment]
Big :thumbsup: -- in future PR review, we'll have to keep a close eye on synchronization/drift between the two.

**Pull request merged**